# High Point Sofa and Chair

In [1]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

In [10]:
df = pd.read_csv('./HighPointSofaandChair.csv')
df.head()

Worker  Performance  Aptitude  Union
0    Abbott           58         5      0
1  Anderson           53         4      0
2    Bender           33        10      0
3      Bush           97        10      0
4    Center           36         2      0

# Variables independientes
x1,x2

In [3]:
X= df[['Aptitude', 'Union']]
X

Aptitude  Union
0          5      0
1          4      0
2         10      0
3         10      0
4          2      0
5          7      0
6          6      0
7          9      0
8          9      1
9          2      1
10         8      1
11         6      1
12         7      1
13         3      1
14         6      1

# Variable dependiente

In [4]:
y = df[['Performance']]
y

Performance
0            58
1            53
2            33
3            97
4            36
5            83
6            67
7            84
8            98
9            45
10           97
11           90
12           96
13           66
14           82

In [6]:
model = LinearRegression()
model.fit(X, y)
y_pred = model.predict(X)

#Metricas estadisticas
mse = mean_squared_error(y, y_pred) 
r2 = r2_score(y, y_pred) 

In [7]:
model.intercept_

array([29.27618014])

In [9]:
print('Coeficientes: ', model.coef_)
print('Error cuadratico de la media: ', mse)
print('Varianza: ', r2)

Coeficientes:  [[ 5.22246337 22.13510581]]
Error cuadratico de la media:  228.9361548200398
Varianza:  0.5329951510923939


# Pregunta del ejercicio:
¿Son eficaces estas dos variables (aptitude, union) para explicar la variacion del desempeño laboral?
Si son eficaces, pero no perfectas. Esto es porque con estas dos variables el modelo logra explicar alrededor del 53% de la variación del desempeño laboral (r2= 0.5329). 